includes/abstracts/llms-abstract-generator-posts.php

```diff
@@ -0,0 +1,782 @@
+		}
+
+		// Retrieve author ID from raw.
+		$author_id = $this->get_author_id_from_raw( $raw, $author_id );
+
+		// Insert the object.
+		$post = new $class_name(
+			'new',
+			array(
+				'post_author'   => $author_id,
+				'post_content'  => isset( $raw['content'] ) ? $raw['content'] : '',
+				'post_date'     => isset( $raw['date'] ) ? $this->format_date( $raw['date'] ) : null,
+				'post_modified' => isset( $raw['modified'] ) ? $this->format_date( $raw['modified'] ) : null,
+				'post_status'   => isset( $raw['status'] ) ? $raw['status'] : $this->get_default_post_status(),
+				'post_title'    => $raw['title'],
+			)
+		);
+
+		if ( ! $post->get( 'id' ) ) {
+			// Translators: %s = post type name.
+			throw new Exception( sprintf( __( 'Error creating the %s post object.', 'lifterlms' ), $type ), self::ERROR_CREATE_POST );
+		}
+
+		// Store the temp id if it exists.
+		$this->store_temp_id( $raw, $post );
+
+		// Don't set these values again.
+		unset( $raw['id'], $raw['author'], $raw['content'], $raw['date'], $raw['modified'], $raw['name'], $raw['status'], $raw['title'] );
+
+		// Set featured image.
+		if ( isset( $raw['featured_image'] ) ) {
+			$this->set_featured_image( $raw['featured_image'], $post->get( 'id' ) );
+		}
+
+		// Set all metadata.
+		foreach ( array_keys( $post->get_properties() ) as $key ) {
+			if ( isset( $raw[ $key ] ) ) {
+				$post->set( $key, $raw[ $key ] );
+			}
+		}
+
+		// Add custom meta.
+		$this->add_custom_values( $post->get( 'id' ), $raw );
+
+		// Handle "extras".
+		$this->sideload_images( $post, $raw );
+		$this->handle_reusable_blocks( $post, $raw );
+
+		return $post;
+
+	}
+
+	/**
+	 * Creates a reusable block
+	 *
+	 * @since [version]
+	 *
+	 * @param int   $block_id WP_Post ID of the block being imported. This will be the ID as found on the original site.
+	 * @param array $block    {
+	 *     Array of block data.
+	 *
+	 *     @type string $title   Title of the reusable block.
+	 *     @type string $content Content of the reusable block.
+	 * }
+	 * @return false|WP_Error|int Returns `false` when the block already exists, an error object if an error is encountered creating the block, or
+	 *                            the new block's WP_Post ID as an integer on success.
+	 */
+	protected function create_reusable_block( $block_id, $block ) {
+
+		$block_id = absint( $block_id );
+
+		// If the block already exists, don't create it again.
+		$existing = get_post( $block_id );
+		if ( $existing && 'wp_block' === $existing->post_type && $block['title'] === $existing->post_title && $block['content'] === $existing->post_content ) {
+			return false;
+		}
+
+		// Check if the block was previously imported.
+		$id = empty( $this->reusable_blocks[ $block_id ] ) ? false : $this->reusable_blocks[ $block_id ];
+		if ( ! $id ) {
+
+			$id = wp_insert_post(
+				array(
+					'post_content' => $block['content'],
+					'post_title'   => $block['title'],
+					'post_type'    => 'wp_block',
```

Comment:
```suggestion
					'post_type'    => 'wp_block',
					'post_status'  => 'publish',
```
otherwise the block will be created as 'draft' and you'll just reference a block that technically "doesn't exist" (that's how gutenberg sees it).

Reply:
good call